core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -117,6 +118,17 @@ public synchronized void updateDynamicFiltersVersionAndFetchIfNecessary(long new
         fetchDynamicFiltersIfNecessary();
     }
 
+    private synchronized void stop()
+    {
+        running = false;
```

Comment:
let's convert this to a state enum, so that it is not possible to start() after stop()

(ok as a follow up)

Reply:
Stop is not possible before start because stop can only be called from request handler